### ISSUE TITLE
Text Search

### ISSUE THREAD

**Issue by lreading** [OWASP/threat-dragon#194]
**Describe what problem your feature request solves**
Ability to search for elements based on user-defined text (element type, description, etc)

**Describe the solution you'd like**
When viewing a diagram, I'd like a search option to be able to find only UI elements that include the given search term. 

**Additional context**
To start with, this can be a simple text search that searches all element types and all fields but expanded on later for more "advanced" search features (only certain types, fields, etc).

This improvement was suggested by Simon S 

It would be good to be able to search for any text within the model
Note that the search box has been temporarily removed via merge #502 
linked to issue #454

**Comment by jgadsden** [comment]
Bumping this to version-2.x, as it is an enhancement that can wait till after the release of version 2.0

**Comment by jgadsden** [comment]
The Search Text box supplied in the left side panel of version 2.0 has been temporarily removed in pull request #502
This can be reapplied once the search feature is fully functional

**Comment by shikida** [comment]
It would be nice if it could be like Mural.co. The user searches for a term and it returns the diagram elements in a list. When the user clicks on the element in the list, the UI focus on the diagram element.

**Comment by jgadsden** [comment]
I agree @shikida , and if there is any help you can give on this feature that would be great
